<?xml version="1.0"?>

<launch>


	<include file="$(find carina_description)/launch/carina_new_model.launch" />

 	
	<!-- send carina.urdf to param server -->
	<!-- 
	<param name="robot_description" textfile="$(find rosbombeiros_simulator)/objects/carina.urdf" /> 
	<param name="robot_description" textfile="$(find carina_description)/urdf/carina.urdf" /> 
	<param
		name="robot_description"
		textfile="$(find rosbombeiros_simulator)/objects/model.sdf" />

	<param
		name="robot_description"
		textfile="$(find carina_description)/urdf/carina_gazebo.urdf" /> 
	
	<param
		name="robot_description"
		textfile="$(find rosbombeiros_simulator)/objects/model.sdf" /> 
	
	<param
		name="tree_description"
		textfile="$(find rosbombeiros_simulator)/objects/tree_a.urdf" /> 
		
	<param
		name="box_description"
		textfile="$(find rosbombeiros_simulator)/objects/box.urdf" /> 
		
	<param
		name="r2d2_description"
		textfile="$(find rosbombeiros_simulator)/objects/r2d2.urdf" /> 
	-->
	
	<!-- push robot_description to factory and spawn robot in gazebo -->
	<node
		name="carina_object"
		pkg="gazebo"
		type="spawn_model"
		args="-urdf -param robot_description -x 0.0 -y 0.0 -z 0.05 -model carina"
		respawn="false"
		output="screen" />

	<!-- 
	<node
		name="joint_state_publisher"
		pkg="joint_state_publisher"
		type="joint_state_publisher"
		respawn="false"
		output="screen" />
 		
	<node
		name="robot_state_publisher"
		pkg="robot_state_publisher"
		type="state_publisher"
		respawn="false"
		output="screen" />
	 -->
	 
	<!--- run EKF filter 
	<node
		pkg="robot_pose_ekf"
		type="robot_pose_ekf"
		name="robot_pose_ekf"
	>
		<remap
			from="imu_data"
			to="imu_gazebo" />
		<remap
			from="odom"
			to="odom_gazebo" />
		<remap
			from="vo"
			to="odom_scan" />

		<param
			name="output_frame"
			value="odom" />
		<param
			name="freq"
			value="10.0" />
		<param
			name="sensor_timeout"
			value="1.0" />
		<param
			name="publish_tf"
			value="true" />
		<param
			name="odom_used"
			value="true" />
		<param
			name="imu_used"
			value="true" />
		<param
			name="vo_used"
			value="false" />
	</node>
	-->
	
	<rosparam
		command="load"
		file="$(find carina_controller)/carina_controller.yaml" />
	<node
		name="carina_controller"
		pkg="pr2_controller_manager"
		type="spawner"
		args="carina_controller"
		respawn="false"
		output="screen" />

</launch>
